File: app/src/main/java/denis/easyweather/app/router/Screen.java
package denis.easyweather.app.router;

import android.os.Bundle;

public class Screen {

    public final Type type;
    public final Bundle arguments;

    public Screen(Type type, Bundle arguments) {
        this.type = type;
        this.arguments = arguments;
    }

    public static Screen fromType(Type type) {
        return new Screen(type, new Bundle());
    }

    public enum Type {
        MAIN,
        DETAIL
    }
}

File: app/src/main/java/denis/easyweather/app/router/Screen.kt
package denis.easyweather.app.router

import android.os.Bundle

class Screen(val type: Type, val arguments: Bundle) {

    enum class Type {
        MAIN,
        DETAIL
    }

    companion object {

        fun fromType(type: Type): Screen {
            return Screen(type, Bundle())
        }
    }
}
